src/main/java/fr/insee/publicenemy/api/infrastructure/csv/SurveyUnitCsvService.java

```diff
@@ -60,4 +59,31 @@ private SurveyUnit getSurveyUnit(@NonNull SurveyUnitCsvLine surveyUnitCsvLine, S
         surveyUnitData.addAttribute("IdUe", surveyUnitId);
         return new SurveyUnit(surveyUnitId, questionnaireModelId, surveyUnitData, SurveyUnitStateData.createInitialStateData());
     }
+
+    @Override
+    public SurveyUnitCsvHeaderLine getSurveyUnitsCsvHeaders(List<VariableType> variablesType) {
+        Set<String> csvHeaders = new LinkedHashSet<>();
+        csvHeaders.add("IdUe");
+        variablesType.forEach(
+                variableType -> csvHeaders.addAll(getCsvHeaders(variableType)));
+        return new SurveyUnitCsvHeaderLine(csvHeaders);
+    }
+
+    /**
+     * Get CSV Headers for a specific variable type
+     * @param variableType variable type
+     */
+    private Set<String> getCsvHeaders(VariableType variableType) {
+        Set<String> csvHeaders = new LinkedHashSet<>();
+        if(!variableType.hasMultipleValues()) {
+            csvHeaders.add(variableType.name());
+            return csvHeaders;
+        }
+
+        // if variable type is an iteration, generate 2 headers
+        for(int index = 1; index <= 2; index++ ) {
+            csvHeaders.add(variableType.name() + "_" + index);
```

Comment:
Avec ce genre de convention je m'inquiète toujours du cas où on aurait une variable hors itération nommée "FOO_1", et une variable dans une itération nommée "FOO" (rien ne l'interdit a priori il me semble).

Reply:
Rien ne l'interdit, c'est un point que j'ai déjà soulevé avec le métier. C'est le format CSV utilisé également dans d'autres process. A réfléchir ...